src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -0,0 +1,305 @@
+     * @param unknown - what JsonElement are we trying to cast
+     * @return Object - the expected object, properly cast
+     */
+    default Object castObject(Object known, JsonElement unknown) {
+        Object objectVal;
+        try {
+            if (known instanceof String) {
+                objectVal = unknown.getAsString();
+            } else if (known instanceof Integer) {
+                objectVal = unknown.getAsInt();
+            } else if (known instanceof Double) {
+                objectVal = unknown.getAsDouble();
+            } else if (known instanceof Float) {
+                objectVal = unknown.getAsFloat();
+            } else if (known instanceof Long) {
+                objectVal = unknown.getAsLong();
+            } else if (known instanceof Boolean) {
+                objectVal = unknown.getAsBoolean();
+            } else if (known instanceof Byte) {
+                objectVal = unknown.getAsByte();
+            } else if (known instanceof Character) {
+                objectVal = unknown.getAsCharacter();
+            } else if (known instanceof JsonArray) {
+                objectVal = unknown.getAsJsonArray();
+            } else if (known instanceof JsonObject) {
+                objectVal = unknown.getAsJsonObject();
+            } else {
+                objectVal = unknown;
+            }
+            return objectVal;
+        } catch (UnsupportedOperationException | NumberFormatException | IllegalStateException e) {
+            return unknown;
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    // checks for our services
+    ///////////////////////////////////////////////////////////////////
+
+    /**
+     * Checks the actual response code is equals to the expected response
+     * code, and writes that out to the output file
+     *
+     * @param expectedCode - the expected response code
+     */
+    void equals(int expectedCode);
+
+    /**
+     * Checks the actual response code is equals to the expected response
+     * code, and writes that out to the output file
+     *
+     * @param expectedCode - the expected response code
+     */
+    default int checkEquals(int expectedCode) {
+        int actualCode = getResponse().getCode();
+        if (actualCode == expectedCode) {
+            getReporter().pass("", "Expected to find a response code of <b>" + expectedCode + "</b>", "Found a response code of <b>" + actualCode + "</b>");
+        } else {
+            getReporter().fail("", "Expected to find a response code of <b>" + expectedCode + "</b>", "Found a response code of <b>" + actualCode + "</b>");
+        }
+        return actualCode;
+    }
+
+    /**
+     * Checks the actual response json payload is equal to the expected
+     * response json payload, and writes that out to the output file
+     *
+     * @param expectedJson - the expected response json object
+     */
+    void equals(JsonObject expectedJson);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either override Object.equals(Object), or rename the method to prevent any confusion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1201)